walk/config.go

```diff
@@ -151,6 +172,50 @@ func loadBazelIgnore(repoRoot string) (isIgnoredFunc, error) {
 	return isIgnored, nil
 }
 
+type gitignoreEntry struct {
+	matcher  gitignore.Matcher
+	patterns []gitignore.Pattern
+}
+
+func createGitIgnorer(parent *gitignoreEntry, rel string, ignoreReader io.Reader) *gitignoreEntry {
+	var domain []string
+	if rel != "" {
+		domain = strings.Split(rel, "/")
+	}
+
+	var matcherPatterns []gitignore.Pattern
+
+	reader := bufio.NewScanner(ignoreReader)
+	for reader.Scan() {
+		// Trim, ignore empty lines and comments
+		i := strings.TrimSpace(reader.Text())
+		if i == "" || strings.HasPrefix(i, "#") {
+			continue
+		}
+
+		matcherPatterns = append(matcherPatterns, gitignore.ParsePattern(i, domain))
+	}
+
+	// Fallback to the parent if no patterns are found
```

Comment:
so this is the case when you have an empty `.gitignore`? That seems rare enough that I would replace this check and the `if` below with just `matcherPatterns = append(parent.patterns, matcherPatterns...)` - I think this should work even when `parent` is nil